### ISSUE TITLE
`space-before-function-paren` doesn't detect anonymously exported function correctly

### ISSUE THREAD

**Issue by Ky6uk** [palantir/tslint#3040]
### Bug Report

- __TSLint version__: `v5.5.0`
- __TypeScript version__: `v2.4.1`
- __Running TSLint via__: Atom / CLI

#### TypeScript code being linted

```ts
export default function () {
  return;
}
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:latest",

  "rules": {
    "ordered-imports": false,
    "object-literal-sort-keys": false,
    "only-arrow-functions": false,

    "interface-name": [true, "never-prefix"],
    "quotemark": [true, "single", "jsx-double"],
    "trailing-comma": [true, "never"],

    "space-before-function-paren": [true, {
      "anonymous": "always",
      "asyncArrow": "always",
      "constructor": "never",
      "method": "never",
      "named": "never"
    }]
  }
}
```

#### Actual behavior
Got the error `Spaces before function parens are disallowed`.

#### Expected behavior
No error expected here.

**Comment by ajafff** [comment]
exported function declarations cannot be anonymous, the grammar doesn't allow that. You should get an error from the typescript compiler. That's why tslint treats this function as `"named"`.

**Comment by Ky6uk** [comment]
@ajafff but this is correct TS code with some compiler flags. I think TSLint should handle that. Correct me if I am wrong.

**Comment by ajafff** [comment]
What flag(s) do you mean? I haven't noticed anything like this before...

**Comment by Ky6uk** [comment]
Sorry. That's working w/o any flags. I've update my first message.

https://www.typescriptlang.org/docs/handbook/modules.html

```
Classes and function declarations can be authored directly as default exports. Default export class and function declaration names are optional.
```

**Comment by ajafff** [comment]
Yeah, the default export makes sense. This is a bug. It should be handled like other anonymous functions.

**Comment by IllusionMH** [comment]
Looks like I found fix, but can't run `yarn test` on Windows at this moment.
Will send PR in the evening.
